### PR TITLE
feat: US133817 Add attempts restrictions

### PR DESCRIPTION
**Pull request by martinwang-d2l**
US133817: 

- Added support for checking if attempt restrictions is supported in mobx. 
- Hide attempt restrictions checkbox and items under checkbox's indentation based on whether attempt restrictions is supported.

### CODE REVIEW

**Comment by martinwang-d2l** [comment]
https://rally1.rallydev.com/#/29180338367d/custom/486232622040?detail=%2Fuserstory%2F616680095557

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-access-restrictions-editor.js [hunk 1]
```diff
@@ -120,33 +118,34 @@ class ActivityQuizSubmissionViewsAccessRestrictionsEditor
...
-				<label class="d2l-label-text">
-					${this.localize('submissionViewDialogSecondaryEditorAttemptLabel')}
-				</label>
-				${this._generateAttemptRestrictionDropdownComponent(entity)}
+			${isAttemptRestrictionsSupported ? html`
+				<div class="d2l-activity-quiz-submission-view-restrict-access-label-container">
+					<label class="d2l-label-text">
+						${this.localize('submissionViewDialogSecondaryEditorRestrictAccessLabel')}
+					</label>
+				</div>
```
Comment (martinwang-d2l) [inline components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-access-restrictions-editor.js:126]: The gradeRestrictions and attemptRestrictions checkboxes are unchanged - the two checkboxes still exist: it's just git diff's formatting thinking it's overwritten.

**Review by mostlyfabulous — APPROVED** [review]
Any idea what caused the whitespace change?

**Comment by martinwang-d2l** [comment]
> Any idea what caused the whitespace change?

For the whitespace in the CSS code I removed it to make it more consistent

**Pull request merged**